src/deriving.ml

```diff
@@ -253,7 +253,11 @@ module Generator = struct
       List.concat lerr
       |> List.map ~f:(fun err -> ext_to_item ~loc:Location.none err [])
     in
-    List.concat l1 @ [ { items = lerr; unused_code_warnings = false } ]
+    List.concat l1
+    @
+    match lerr with
+    | [] -> []
+    | _ :: _ as lerr -> [ { items = lerr; unused_code_warnings = false } ]
```

Comment:
@NathanReb my reasoning here was :

1. Inserting `unused_code_warnings=false` impacts what the new code does since after these items are merged with the rest of the one generated by a ppx, `types_used_by_deriving` doesn't distinguish it from ppx generated ones, and believes one of the ppx intends not to allow the warnings. Thus, it is better to omit this extra element in the list if the `lerr` list to be inserted is empty (and result in the same generated items overall).

2. In case `lerr` is not empty, (this part I am more fuzzy on), I am guessing this is a mechanism designed in ppxlib to avoid stopping on the first error reported by one ppx. I made the hypothesis that in this case the entire ppxchain is going to be aborted in the end due to processing errors, and thus the impact of `unused_code_warnings` can be ignored in this code path (when errors are emitted).

Reply:
I would suggest a slightly different approach here. I think it makes things clearer if we determine whether we should silence warning 34 from the list of generators rather than from the wrapped generated items.

Getting it directly from the field that determines whether they opted out is safer than getting it from a field that derives from that. I can imagine a scenario where changes in the code could inadvertently break this behaviour, as suggests this specific change for instance.

Does that sound good to you?

Reply:
Oh, of course yes. I missed that. Thanks a lot! Done.